datacube_alchemist/worker.py

```diff
@@ -66,14 +66,20 @@ def transform_name(self) -> str:
         return self.config.specification.transform
 
     @property
-    def _transform(self) -> Type[Transformation]:
+    def transform(self) -> Type[Transformation]:
 
         module_name, class_name = self.transform_name.rsplit(".", maxsplit=1)
         module = importlib.import_module(name=module_name)
         imported_class = getattr(module, class_name)
         assert issubclass(imported_class, Transformation)
         return imported_class
 
+    def _native_resolution(self, task: AlchemistTask) -> float:
```

Comment:
I don't like this since everywhere else we have resolution as a tuple of y,x resolutions and not a single floating point number.

So I would return `geobox.resolution` out of this and then if you need a single number for ground sampling distance call `min` or `max` on that.

Reply:
Also to me this feels like a task for the same code that builds `grids`, so `dataset_assembler`. How is this field populated for ARD products, manually from configuration or by `dataset_assembler`?

Reply:
Not sure how it's done in EO Datasets, @jeremyh ?

Reply:
Looks like it's done fairly manually for things like WAGL:
```
    del p.properties["eo:gsd"]
    p.properties["eo:gsd"] = min(
        min(h5group[grp].attrs["resolution"]) for grp in resolution_groups
    )
```

From: https://github.com/GeoscienceAustralia/eo-datasets/blob/eodatasets3/eodatasets3/wagl.py#L174

Reply:
ok, but maybe common well-defined STAC properties, like `eo:gsd`, should be the responsibility of the assembler, to ensure that it's compatible with the `grids`. The logic could be the same as here: `min(min(grid.geobox.resolution) for grid in grids)`